pkg/ha/state.go

```diff
@@ -0,0 +1,58 @@
+package ha
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type State struct {
+	leader          string
+	leaseStart      time.Time
+	leaseUntil      time.Time
+	maxTimeSeen     time.Time // max data time seen by any instance
+	maxTimeInstance string    // the instance name that’s seen the maxtime
+	_mu             sync.RWMutex
+}
+
+type StateView struct {
+	leader          string
+	leaseStart      time.Time
+	leaseUntil      time.Time
```

Comment:
I think we need to add a maxTimeSeenLeader here. This will help us advance leaseUntil in the heartbeater.